source/MultiGrid/MultiGrid.js

```diff
@@ -99,15 +100,15 @@ export default class MultiGrid extends Component {
     const { columnWidth, fixedColumnCount, fixedRowCount, rowHeight } = this.props
 
     if (
-      columnWidth !== nextProps.columnWidth ||
+      columnWidth instanceof Function || columnWidth !== nextProps.columnWidth ||
       fixedColumnCount !== nextProps.fixedColumnCount
     ) {
       this._leftGridWidth = null
     }
 
     if (
       fixedRowCount !== nextProps.fixedRowCount ||
-      rowHeight !== nextProps.rowHeight
+      rowHeight instanceof Function || rowHeight !== nextProps.rowHeight
```

Comment:
The previous behavior here was intentional. If a function prop is passed, `Grid` (and so `MultiGrid`) don't assume they can detect when it has changed. That's because (a) things outside of the function may invalidate the size without the function prop itself changing and (b) people often define inline props like so: `rowHeight={(index) => blah}` which will recreate on each render. Invalidating cache is expensive so this is an intentional optimization. If a user uses a function prop, he should manually clear the cache.